utils/dockerfile_generator.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+import os
+from json import loads
+from sys import argv
+
+
+if __name__ == "__main__":
+
+    root = Path(argv[0]).absolute().parent.parent
+
+    analyzers = [x for x in sorted(Path(root, 'analyzers').iterdir()) if x.is_dir()]
+
+    for analyzer in analyzers:
+        try:
+            dockerfile_path = analyzer / 'Dockerfile'
+
+            update = dockerfile_path.exists()
+
+            if update:
+                with dockerfile_path.open() as dockerfile:
+                    dockerfile = dockerfile.read()
+
+                    # Dockerfiles with this string will be frozen; skip.
+                    if '### MANUAL ###' in dockerfile:
```

Comment:
If you wanna freeze a Dockerfile in time, add this string anywhere (preferably the first line for clarity).

Used this as a compromise for when an analyzer is just too niche to build programatically.